lib/OptimizationNLPModels/src/OptimizationNLPModels.jl

```diff
@@ -0,0 +1,38 @@
+module OptimizationNLPModels
+
+using Reexport
+@reexport using NLPModels, Optimization, ADTypes
+
+function SciMLBase.OptimizationFunction(
+        nlpmodel::AbstractNLPModel; kwargs...)
+    f(x, p) = NLPModels.obj(nlpmodel, x)
+    grad(G, u, p) = NLPModels.grad!(nlpmodel, u, G)
+    hess(u, p) = NLPModels.hess(nlpmodel, u)
+    hv(Hv, u, v, p) = NLPModels.hprod!(nlpmodel, u, v, Hv)
+    cons(res, x, p) = NLPModels.cons!(nlpmodel, x, res)
+    cons_j(x, p) = NLPModels.jac(nlpmodel, x)
```

Comment:
Why are some of these not inplace while others are, I might be missing something... (see the comment above in the `hess`)

Reply:
I tried to make them all in-place, but I didn't find them in the documentation. I explored the source code briefly and didn't see them either so I tried to use as many in-place versions as I could find but some apparently don't exist

Reply:
Have suggested a workaround for now above